dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/deduplication/DefaultDeduplicationService.java

```diff
@@ -261,7 +262,17 @@ private boolean sameAttributesAreEquals( Set<TrackedEntityAttributeValue> tracke
     @Override
     @Transactional
     public void addPotentialDuplicate( PotentialDuplicate potentialDuplicate )
+        throws PotentialDuplicateConflictException
     {
+        if ( !potentialDuplicate.getStatus().isForCreation() )
+        {
+            throw new PotentialDuplicateConflictException(
+                "Potential Duplicate creation must have one of following status : "
```

Comment:
`"Potential Duplicate creation must have one of following status : "`

maybe

`"Invalid status %s, creating potential duplicate is allowed using: "`

replacing the `%s` with the one that was set. Adding the actual value gives more context (for example to a stacktrace).